board/libc.h

```diff
@@ -5,6 +5,15 @@ void delay(uint32_t a) {
   for (i = 0; i < a; i++);
 }
 
+void delay_us(uint32_t a) {
+  uint32_t end = MICROSECOND_TIMER->CNT + a;
+  while(MICROSECOND_TIMER->CNT < end);
+}
+
+void delay_ms(uint32_t a) {
+  delay_us(a*1000U);
+}
+
```

Comment:
This looks like a great place for edge cases, and I'm pretty sure this doesn't work.
Example: counter is 50 away from overflowing. a is set to 100, such that 'end' is already overflown. The while loop will exit immediately.
This also relies on the fact that nobody will set the auto reset value of the timer and that it's a 32 bits one.

<s>Also: do we have enough timers that we can assign one to this?</s> wasn't aware we already had this timer